### PR TITLE
Update2016

### PR DESCRIPTION
**Pull request by versokova**
In cl/llvm fixed bugs and added external passes over LLVM IR. Will be better create passes_lib as a git module? Probably no. Added scripts fo LLVM front end and fix building with local version of Clang/LLVM. I don't known how to create register-paths.sh for LLVM, it is the same like for gcc?

Updated SV-COMP check-property.sh script, now support LLVM (for testing). Added modified property for reachability (instead ERROR label), because LLVM don't recognized name of label.

### CODE REVIEW

File: build-aux/xlib.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# common code base for fa_build/fa{gcc,db,llvm} and sl_build/sl{gcc,gdb,llvm}
```
Comment (kdudka) [inline build-aux/xlib.sh]: I would not rename the file as ```xlib.sh``` as one might think it refers to https://en.wikipedia.org/wiki/Xlib -- what about ```cclib.sh``` instead?

Of course, no need to resubmit...

**Comment by kdudka** [comment]
Thank you for working on this!  I will merge it soon hopefully...

As for ```register-paths.sh```, it just exports ```$PATH``` and ```$LD_LIBRARY_PATH``` environment variables.  The former helps to find the compiler executable and the latter helps to find the compiler plug-in (shared library).  Both of them can be built locally and, after sourcing the generated ```register-paths.sh``` script, they can be used as if they were installed on the system.  Such a script was once proposed by PP and it is still advertised on Predator's web page.  I am not sure if anyone actively uses it though.

**Comment by kdudka** [comment]
I have prepared the commits for merge in the ```llvm``` branch:

https://github.com/kdudka/predator/compare/llvm

Please let me know whether you are fine with merging it such.

**Comment by versokova** [comment]
OK

**Comment by kdudka** [comment]
Perfect.  Pushed as https://github.com/kdudka/predator/compare/be9949c...bd9749f.  Thanks for the contribution!

**Pull request closed without merging**